solr/core/src/java/org/apache/solr/servlet/ServletUtils.java

```diff
@@ -252,8 +253,11 @@ private static void traceHttpRequestExecution2(
   static void consumeInputFully(HttpServletRequest req, HttpServletResponse response) {
     try {
       ServletInputStream is = req.getInputStream();
-      //noinspection StatementWithEmptyBody
-      while (!is.isFinished() && is.read() != -1) {}
+      if (!is.isFinished() && is.read() != -1) {
```

Comment:
even the read here isn't needed, technically, but it's super common for an InputStream to not be finished yet the read is EOF.  skipNBytes allocates a 2k array so we can avoid that typically.

Reply:
We have `Utils.readFully(InputStream is)` would that usable here?

Reply:
Ah yes, thanks!
Feedback addressed.